grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java

```diff
@@ -594,4 +596,15 @@ private void processExtension(FieldDescriptor extension, FileDescriptor fd) {
             fileDescriptorsByExtensionAndNumber.get(extensionName).put(extensionNumber, fd);
         }
     }
+
+    private <T> T accessIndex(Supplier<T> operation) {
+        try {
+            indexAccess.acquire();
+            T result = operation.get();
+            indexAccess.release();
+            return result;
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Anytime you catch an `InterruptedException` you have to deal with it; this currently clears the interrupt status.  Better:
```
} catch (InterruptedException e) {
    Thread.currentThread().interrupt(); // re-trigger interrupt status
    thrwo new RuntimeException(e.getMessage(), e);
}
```

Reply:
Fixed